### PR TITLE
Change package manager to PNPM and make CI work

### PR DESCRIPTION
**Pull request by vstefanovic97**

### CODE REVIEW

File: packages/ember-runtime-config/package.json [hunk 1]
```diff
@@ -121,7 +124,7 @@
     "./middleware": "./middleware/index.js"
   },
   "typesVersions": {
-    "*": {
+    ">=5.0.0": {
```
Comment (SergeAstapov) [inline packages/ember-runtime-config/package.json:127]: @vstefanovic97 why do we need this?

File: package.json [hunk 2]
```diff
@@ -29,9 +25,33 @@
...
+          }
+        }
+      }
+    },
+    "peerDependencyRules": {
+      "allowAny": [
+        "ember-runtime-config",
+        "typescript"
+      ],
+      "ignoreMissing": [
```
Comment (SergeAstapov) [inline package.json:47]: @vstefanovic97 why do we need this? where `webpack` is missing as dep?

File: packages/test-app/package.json [hunk 3]
```diff
@@ -82,6 +84,22 @@
     "typescript": "^5.1.6",
     "webpack": "^5.88.1"
   },
+  "dependenciesMeta": {
+    "ember-runtime-config": {
+      "injected": true
```
Comment (SergeAstapov) [inline packages/test-app/package.json:89]: @vstefanovic97 why do we need this to be injected? is this due to some peerDeps? whic ones?

File: packages/ember-runtime-config/package.json (hunk 1 above)
Comment (vstefanovic97) [reply packages/ember-runtime-config/package.json:127]: We use `allowImportingTsExtensions` option which is only supported by version 5 or higher

File: package.json (hunk 2 above)
Comment (vstefanovic97) [reply package.json]: image
I really doubt we should be actually adding webpack as a devDep there

File: packages/test-app/package.json (hunk 3 above)
Comment (vstefanovic97) [reply packages/test-app/package.json]:
```
Some V1 ember addons are resolving as incorrect peer dependencies. This makes it impossible for us to safely convert them to v2 format.

  👇 👇 👇
👉 See https://github.com/embroider-build/embroider/blob/main/docs/peer-dependency-resolution-issues.md for an explanation of the problem and suggestions for fixing it.
  👆 👆 👆

test-app@1.0.1 (dev)-> ember-runtime-config@1.0.1
    sees peerDep @ember/test-helpers@3.3.1
      at /home/runner/work/ember-runtime-config/ember-runtime-config/node_modules/.pnpm/@ember+test-helpers@3.3.1_@babel+core@7.25.2_@glint+template@1.4.0_ember-source@5.10.2_@glimm_ij6vgliezcq6yn5qud3sz4bngy/node_modules/@ember/test-helpers
    but test-app@1.0.1 is using @ember/test-helpers@3.3.1
      at /home/runner/work/ember-runtime-config/ember-runtime-config/node_modules/.pnpm/@ember+test-helpers@3.3.1_@babel+core@7.25.2_@glint+template@1.4.0_ember-source@5.2.0_@babel+_ayvlfwcg5saapuu4dpj5wq2way/node_modules/@ember/test-helpers

test-app@1.0.1 (dev)-> ember-runtime-config@1.0.1
    sees peerDep ember-source@5.10.2
      at /home/runner/work/ember-runtime-config/ember-runtime-config/node_modules/.pnpm/ember-source@5.10.2_@glimmer+component@1.1.2_@babel+core@7.25.2__@glint+template@1.4.0_rsvp@4.8.5/node_modules/ember-source
    but test-app@1.0.1 is using ember-source@5.2.0
      at /home/runner/work/ember-runtime-config/ember-runtime-config/node_modules/.pnpm/ember-source@5.2.0_@babel+core@7.25.2_@glimmer+component@1.1.2_@babel+core@7.25.2__@glint+tem_2g3agtudzhcih6ftpprlnff2sm/node_modules/ember-source
```

It happens for embroider builds when working with ember-try in CI.
Found that inject deps meta as a solution, saw some other ember repos using it

File: package.json (hunk 2 above)
Comment (SergeAstapov) [reply package.json:47]: @vstefanovic97 thank you for explanation!

**Review by SergeAstapov — APPROVED** [review]

**Pull request merged**